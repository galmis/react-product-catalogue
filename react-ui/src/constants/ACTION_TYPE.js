// @flow

export const FILTER = 'FILTER';
export const FETCH_POSTS = 'FETCH_POSTS';
export const FETCH_POSTS_SUCCESS = 'FETCH_POSTS_SUCCESS';
export const FETCH_COMMENTS = 'FETCH_COMMENTS';
export const FETCH_COMMENTS_SUCCESS = 'FETCH_COMMENTS_SUCCESS';
export const FETCH_DATA_ERROR = 'FETCH_DATA_ERROR';
