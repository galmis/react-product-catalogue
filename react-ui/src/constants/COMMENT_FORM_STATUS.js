// @flow

import { SUCCESS, WARNING, ERROR } from './COMMENT_FORM_STATUS_TYPE';

const COMMENT_FORM_STATUS = Object.freeze({
  ['201']: {
    type: SUCCESS,
    code: 201,
    message: 'Komentaras išsiųstas sėkmingai.',
  },
  ['400']: {
    type: ERROR,
    code: 400,
    message: 'Komentaras neišsiųstas. Palaukite 15 sekundžių ir bandykite dar kartą.'
  },
  ['409']: {
    type: ERROR,
    code: 409,
    message: 'Komentaras neišsiųstas. Jūsų komentaras su tokiu pačiu tekstu jau egzistuoja.'
  }
});

export default COMMENT_FORM_STATUS;
