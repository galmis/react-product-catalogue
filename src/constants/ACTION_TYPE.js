// @flow

const ACTION_TYPE: Object = Object.freeze({

});

export default ACTION_TYPE;
