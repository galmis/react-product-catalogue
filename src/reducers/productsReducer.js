// @flow

const initialState = {
  allIds: ['0', '1', '2', '3', '4'],
  byId: {
    '0': {
      id: '0',
      title: 'Antiox',
      thumbText: 'Aha.. Antioxas gerai!',
      htmlText: '<p><script>alert("XSS attack MUHAHA!");</script> Valio Antiox! </p>'
    },
    '1': {
      id: '1',
      title: 'Detox',
      thumbText: 'Laba diena su vistiena',
      htmlText: '<p> Valio Detox! </p>'
    },
    '2': {
      id: '2',
      title: 'Pax Forte',
      thumbText: 'Liurbis Antanas',
      htmlText: '<p> Valio Pax! </p>'
    },
    '3': {
      id: '3',
      title: 'HeparD',
      thumbText: 'Gera diena...',
      htmlText: '<p> Valio HeparD! </p>'
    },
    '4': {
      id: '4',
      title: 'Cardio Drive',
      thumbText: 'Cardio vairavims... kon to?',
      htmlText: '<p> Valio Cardio! </p>'
    }
  }
};

// for now always returns initialState, but this might change
export default function productsReducer(state: Object = initialState, action: Object) {

  return state;
}
