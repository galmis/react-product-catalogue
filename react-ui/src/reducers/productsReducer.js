// @flow

import { FILTER } from '../constants/ACTION_TYPE';

const initialState = {
  selectedCategory: 'Visi',
  allIds: ['0', '1', '2', '3', '4', '5'],
  byId: {
    '0': {
      id: '0',
      title: 'Antiox+',
      thumbText: '',
      articleFile: 'Antiox.js',
      imgFile: 'antiox-crop.jpg',
      categories: ['Visi', 'Klasikiniai']
    },
    '1': {
      id: '1',
      title: 'Detox+',
      thumbText: 'Imuniteto stiprinimui, organizmo išvalymui',
      articleFile: 'Detox.js',
      imgFile: 'detox-crop.jpg',
      categories: ['Visi', 'Klasikiniai']
    },
    '2': {
      id: '2',
      title: 'Pax+ forte',
      thumbText: 'Kasdieninė apsauga nuo streso',
      articleFile: 'PaxForte.js',
      imgFile: 'pax-forte-crop.jpg',
      categories: ['Visi', 'Klasikiniai']
    },
    '3': {
      id: '3',
      title: 'Sveltform+',
      thumbText: 'Kaip normalizuoti medžiagų apykaitą ir mažinti svorį?',
      articleFile: 'Sveltform.js',
      imgFile: 'sveltform-crop.jpg',
      categories: ['Visi', 'Klasikiniai']
    },
    '4': {
      id: '4',
      title: 'Chromevital+',
      thumbText: 'Energijos užtaisas kiekvienai ląstelei',
      articleFile: 'Chromevital.js',
      imgFile: 'chromevital-crop.jpg',
      categories: ['Visi', 'Klasikiniai']
    },
    '5': {
      id: '5',
      title: 'CardioDrive',
      thumbText: 'Visapusė širdies sveikata',
      articleFile: 'CardioDrive.js',
      imgFile: 'cardio-drive-crop.jpg',
      categories: ['Visi', 'Nauji']
    },
  }
};

export default function productsReducer(state: Object = initialState, action: Object) {

  switch(action.type) {
    case FILTER: {
      return {
        ...state,
        selectedCategory: action.payload.category
      }
    }
  }

  return state;
}
