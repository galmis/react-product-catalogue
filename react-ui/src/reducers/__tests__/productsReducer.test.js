// @flow
'use strict';

jest.dontMock('../productsReducer.js');

import productsReducer from '../productsReducer';
import deepFreeze from 'deep-freeze';

import { FILTER } from '../../constants/ACTION_TYPE';

describe('productsReducer', () => {

  it('FILTER, initialState', () => {
    const action = {
      type: FILTER,
      payload: {
        category: 'foo'
      }
    };
    deepFreeze(action);

    expect(productsReducer(undefined, action)).toMatchSnapshot();
  });

});
