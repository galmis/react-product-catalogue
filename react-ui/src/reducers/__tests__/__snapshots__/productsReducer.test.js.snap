exports[`productsReducer FILTER, initialState 1`] = `
Object {
  "allIds": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
  ],
  "byId": Object {
    "0": Object {
      "articleFile": "Antiox.js",
      "categories": Array [
        "Visi",
        "Klasikiniai",
      ],
      "id": "0",
      "imgFile": "antiox-crop.jpg",
      "thumbText": "",
      "title": "Antiox+",
    },
    "1": Object {
      "articleFile": "Detox.js",
      "categories": Array [
        "Visi",
        "Klasikiniai",
      ],
      "id": "1",
      "imgFile": "detox-crop.jpg",
      "thumbText": "Imuniteto stiprinimui, organizmo išvalymui",
      "title": "Detox+",
    },
    "2": Object {
      "articleFile": "PaxForte.js",
      "categories": Array [
        "Visi",
        "Klasikiniai",
      ],
      "id": "2",
      "imgFile": "pax-forte-crop.jpg",
      "thumbText": "Kasdieninė apsauga nuo streso",
      "title": "Pax+ forte",
    },
    "3": Object {
      "articleFile": "Sveltform.js",
      "categories": Array [
        "Visi",
        "Klasikiniai",
      ],
      "id": "3",
      "imgFile": "sveltform-crop.jpg",
      "thumbText": "Kaip normalizuoti medžiagų apykaitą ir mažinti svorį?",
      "title": "Sveltform+",
    },
    "4": Object {
      "articleFile": "Chromevital.js",
      "categories": Array [
        "Visi",
        "Klasikiniai",
      ],
      "id": "4",
      "imgFile": "chromevital-crop.jpg",
      "thumbText": "Energijos užtaisas kiekvienai ląstelei",
      "title": "Chromevital+",
    },
    "5": Object {
      "articleFile": "CardioDrive.js",
      "categories": Array [
        "Visi",
        "Nauji",
      ],
      "id": "5",
      "imgFile": "cardio-drive-crop.jpg",
      "thumbText": "Visapusė širdies sveikata",
      "title": "CardioDrive",
    },
  },
  "selectedCategory": "foo",
}
`;
